plugins/action/action.h

```diff
@@ -24,29 +38,164 @@ class Action
+
+    /**
+     * Commands the drone to return to the launch (takeoff) position and land there (synchronous).
+     *
+     * This switches the drone into RTL mode which generally means it will rise up to a certain
+     * altitude to clear any obstacles before heading back to the launch (takeoff) position and
+     * land there.
+     *
+     * @return result of request
+     */
     Result return_to_launch() const;
 
+    /**
+     * Callback type for async action calls.
+     */
+    typedef std::function<void(Result)> result_callback_t;
+
+    /**
+     * Arms the drone (asynchronous).
+     *
+     * Note that arming a drone normally means that the motors will spin at idle.
+     * Therefore, before arming take all safety precautions and stand clear of the drone.
+     *
+     * @param callback function to call with result of request
+     */
     void arm_async(result_callback_t callback);
+
+    /**
+     * Disarms the drone (asynchronous).
+     *
+     * This will disarm a drone if it considers itself landed. If flying in-air, the drone should
+     * reject the disarm command. Disarming means that all motors will stop.
+     *
+     * @param callback function to call with result of request
+     */
     void disarm_async(result_callback_t callback);
+
+    /**
+     * Kills the drone (asynchronous).
+     *
+     * This will disarm a drone no matter if still landed or still in-air.
+     * Note that the drone will fall out of the sky if you use this command while flying.
+     *
+     * @param callback function to call with result of request
+     */
     void kill_async(result_callback_t callback);
+
+    /**
+     * Commands the drone to take off and hover at a set takeoff altitude (asynchronous).
+     *
+     * This switches the drone into position control mode and commands it to take off and hover at
+     * set takeoff altitude.
+     *
+     * @param callback function to call with result of request
+     */
     void takeoff_async(result_callback_t callback);
+
+    /**
+     * Commands the drone to land at the current position (asynchronous).
+     *
+     * This switches the drone into position control mode and commands it to take off and hover at
+     * set takeoff altitude.
+     *
+     * @param callback function to call with result of request
+     */
     void land_async(result_callback_t callback);
+
+    /**
+     * Commands the drone to return to the launch (takeoff) position and land there (asynchronous).
+     *
+     * This switches the drone into RTL mode which generally means it will rise up to a certain
+     * altitude to clear any obstacles before heading back to the launch (takeoff) position and
+     * land there.
+     *
+     * @param callback function to call with result of request
+     */
     void return_to_launch_async(result_callback_t callback);
 
+    /**
+     * Sets takeoff altitude above ground.
+     *
+     * @param relative_altitude_m takeoff altitude relative to takeoff location in meters
```

Comment:
Again, should this be relative to "home"?

Reply:
I think this is clearer because some think they can reset home which is probably untested and discouraged.